File: converted/utilities/deep-equal/elide-deep-equal.py
#!/usr/bin/env python3
"""
Python Integration Example for elide-deep-equal

Benefits:
- One deep equality implementation shared across TypeScript and Python
- Consistent comparison logic across services
- Handles circular references, Dates, Maps, Sets
- No need for custom Python deep comparison
"""

print("=== Python Calling TypeScript Deep Equal ===\n")

# Example: Deep comparison
# obj1 = {'user': {'name': 'Alice', 'age': 25}}
# obj2 = {'user': {'name': 'Alice', 'age': 25}}
# equal = deep_equal.default(obj1, obj2)
# print(f"Equal: {equal}")  # True

print("Real-world use case:")
print("- Python test suite uses deep equal for assertions")
print("- Uses same TypeScript implementation as Node.js")
print("- Guarantees consistent comparison")
print()

print("When Elide Python API is ready:")
print("  from elide import require")
print("  deep_equal = require('./elide-deep-equal.ts')")
print("  assert deep_equal.default(obj1, obj2)")

File: converted/utilities/deep-equal/elide-deep-equal.rb
#!/usr/bin/env ruby

# Ruby Integration Example for elide-deep-equal

puts "=== Ruby Calling TypeScript Deep Equal ===\n"

# Example: Deep comparison
# obj1 = { user: { name: 'Alice', age: 25 } }
# obj2 = { user: { name: 'Alice', age: 25 } }
# equal = deep_equal_module.default(obj1, obj2)
# puts "Equal: #{equal}"  # true

puts "Real-world use case:"
puts "- Ruby RSpec tests use deep equal for assertions"
puts "- Uses same TypeScript implementation as Node.js"
puts "- No custom deep comparison needed"
puts ""

puts "RSpec Integration (when ready):"
puts "  DEEP_EQUAL = Elide.require('./elide-deep-equal.ts')"
